26-dracula-ant/technical_specification.md

```diff
@@ -0,0 +1,103 @@
+# Implementation of a work package service for CLI client authentication (Dracula Ant)
+
+**Epic Type:** Implementation Epic
+  
+**Attention: Please do not put any confidential content here.**
+
+## Scope:
+
+We want to implement the work package service outlined in
+https://docs.ghga-dev.de/main/architecture_concepts/ac001_file_validation_and_encryption.html#client-configuration-and-authorization
+
+## Implementation Details:
+
+### REST API
+
+The Work Package Service REST API should have the following end points:
+
+- `POST /work_packages`
+  - auth header: internal access token
+  - request body:
+    - `dataset_id`: str
+    - `type`: enum (download/upload)
+    - `file_ids`: array of strings  (empty = all files of the dataset)
+    - `public_key`: string (the user's public Crypt4GH key)
+  - response body:
+    - `id`: the work package id
+    - `token`: work package access token
+- `GET /work_packages`
+  - auth header: internal access token
+  - gets all list of all work packages of the current user
+- `GET /work_packages/{id}`
+  - auth header: internal access token
+  - gets work package with given id if it belongs to the current user
+- `DELETE /work_packages/{id}`
+  - auth header: internal access token
+  - deletes work package with given id if it belongs to the current user
+- `GET /work_package`
+  - auth header: work package access token
+  - gets work package that has the same hash as the given token
+    (note that this is not really RESTful, the work package is passed in directly via the token)
```

Comment:
Can be simplified as the UI does not need to retrieve a work package after creation at the moment:
```suggestion
- `GET /work_packages/{id}`
  - auth header: work package access token
  - gets work package with given id if it belongs to the current user and matches the ID specified in the token
- `DELETE /work_packages/{id}`
  - auth header: internal access token
  - deletes work package with given id if it belongs to the current user
```

Reply:
Simplified, but mention these as possible extension for later.